### PR TITLE
fix: corrected getting of proof request template

### PR DESCRIPTION
**Pull request by Artemkaaas**
# Summary of Changes

Issue: The logic of getting the proof request template was incorrect because the function searched for the request in the default constant. Instead, it must search in the application configuration, so if the app overrides the templates the proper one will be found. 

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [ ] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [ ] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [ ] Updated documentation as needed for changed code and new or modified features;
- [ ] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: packages/legacy/core/App/screens/ProofRequestDetails.tsx [hunk 1]
```diff
@@ -224,10 +224,12 @@ const ProofRequestDetails: React.FC<ProofRequestDetailsProps> = ({ route, naviga
     { visible: boolean; predicate: string | undefined } | undefined
   >(undefined)
 
-  useEffect(() => {
-    const template = getProofRequestTemplate(templateId)
-    if (!template) return
+  const template = useTemplate(templateId)
+  if (!template) {
+    throw new Error('Unable to find proof request template')
```
Comment (jleach) [inline packages/legacy/core/App/screens/ProofRequestDetails.tsx:229]: I think this should be a `BifoldError` - no?

**Review by jleach — CHANGES_REQUESTED** [review]
Just one comment. LMK what you think.

File: packages/legacy/core/App/screens/ProofRequestDetails.tsx (hunk 1 above)
Comment (Artemkaaas) [reply packages/legacy/core/App/screens/ProofRequestDetails.tsx:229]: What is the rule for choosing the error kind?
If the template is not found, it's unexpected behavior as we select it on the previous screen. 
I looked to other screens for credentials and proof  request details and they threw similar errors
```
throw new Error('Unable to fetch proof from AFJ')
throw new Error('Unable to fetch credential from AFJ')
```

**Comment by bryce-mcmath** [comment]
@Artemkaaas I noticed after the verifier PR got merged in, lots of linting issues and a failing test snapshot got through. If you run `npx husky@7.0.0 install` in the root of the repository it will run checks before you commit and before you push. @JamesKEbert not sure if this means our GHA checks aren't working properly

**Review by jleach — APPROVED** [review]

**Comment by jleach** [comment]
@bryce-mcmath Sorry, didn't see your comment above before I approved this one. Must not be running since I think they would pick up this issue.

**Comment by Artemkaaas** [comment]
@bryce-mcmath 
I did know about husky. Why is it not listed in the `devDependencies` of the root package.json?
 
But I regularly ran `yarn lint` and `yarn test` commands manually.  
`yarn lint` listed lots of warnings but they are not caused our changes. `yarn test` passed for me.

**Pull request merged**